server/src/main/java/org/opensearch/indices/replication/common/StatsAwareReplicationCollection.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.replication.common;
+
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.indices.recovery.PeerRecoveryStats;
+import org.opensearch.indices.recovery.PeerRecoveryStatsTracker;
+import org.opensearch.threadpool.ThreadPool;
+
+/**
+ * Wrapper of {@link ReplicationRef} which attaches {@link PeerRecoveryStatsTracker} to track stats of
+ * ongoing relocation.
+ *
+ * @opensearch.internal
+ */
+public class StatsAwareReplicationCollection<T extends ReplicationTarget> extends ReplicationCollection<T> {
+
+    private final PeerRecoveryStatsTracker tracker;
+
+    public StatsAwareReplicationCollection(Logger logger, ThreadPool threadPool) {
+        super(logger, threadPool);
+        this.tracker = new PeerRecoveryStatsTracker();
+    }
+
+    @Override
+    public boolean cancelForShard(ShardId shardId, String reason) {
+        tracker.incrementTotalCancelledRelocation(1);
+        return super.cancelForShard(shardId, reason);
+    }
+
+    @Override
+    public long start(T target, TimeValue activityTimeout) {
+        tracker.incrementTotalStartedRelocation(1);
+        return super.start(target, activityTimeout);
+    }
+
+    @Override
+    public T reset(final long id, final TimeValue activityTimeout) {
+        tracker.incrementTotalRetriedRelocation(1);
```

Comment:
What does reset mean in context of a peer recovery? How does this relate with a retry?